### DISCUSSION TITLE
Scaling external resources feature

### DISCUSSION THREAD

**Discussion by marcportabellaclotet-mt** [kedacore/keda#1814] [Ideas]
I am using keda to autoscale my deployments and it is awesome.
I was wondering if if could be an interesting feature for keda, to allow the possibility to scale external resources like aws databases for example.
This could be useful in some scenarios.
My current use case:
When a deployment scales to 0 pods, stop the aws rds database that the deployment is using. This is saving some money in our dev environments.
Other use cases:
- Vertical| Horitzontal Scale AWS Databases based on metrics.

I have already developed a kubernetes daemon that starts and stops RDS databases when deployments scales ffrom/to 0 (if they are properly annotated)

Probably this is out of the scope of keda, but I just wanted to share the idea.

**Comment by tomkerkhove** [comment]
This is definately a great idea and I've been exploring this myself for Azure, but for this we would typically rely on another product that has a CRD with /scale subresource.

I'm not sure if there are any AWS products that allow you to manage resources in Azure?

**Reply by zroubalik** [reply]
Hi, agree with Tom. I think that the most viable way is to scale through some CustomResource in the middle.

**Reply by marcportabellaclotet-mt** [reply]
Thanks a lot for your comments,
I also agree that this should be managed with another crd, something like externalScaledObject or whatever.
What it would be great, is that the current scalers could manage these externalScaledObjects CRD somehow.
Not sure if this makes sense or not.
@tomkerkhove I am not sure If I understand your question about AWS product to manage Azure resources.

Right now, to scale this external services, I have deployed a go app as a container which scans kubernetes events, and when there is a change in replicas, it applies the logic to manage.
But as I mention, it would be awesome, if all the buitin keda scalers could manage this externalscaledobjects crd.
